lib/vanagon/engine/always_be_scheduling.rb

```diff
@@ -37,7 +37,21 @@ def name
 
       # return the platform name as the "host" name
       def build_host_name
-        @platform.name
+        platform_name_to_vmpooler_style(@platform.name)
+      end
+
+      def platform_name_to_vmpooler_style(platform)
+        translation = {
+          "aix-5.3-ppc" => "aix-53-ppc",
```

Comment:
I'm in favor of having a shared library at some point. I'm not opposed to @joshcooper's idea of `abs_resource_name` (or similar) akin to `vmpooler_template`.  [usages of `vmpooler_template`](https://github.com/search?p=2&q=org%3Apuppetlabs+vmpooler_template&type=Code)